tar/write.c

```diff
@@ -450,6 +450,15 @@ write_archive(struct archive *a, struct bsdtar *bsdtar)
 		archive_entry_linkify(bsdtar->resolver, &entry, &sparse_entry);
 	}
 
+	/*
+	 * Check whether the archive contains 0 bytes; this is unaffected by
+	 * deduplication.  Must be done before the archive is closed.
```

Comment:
s/.$/ since even empty archives end up with nonzero size due to tar EOF blocks./